ui/controls.js

```diff
@@ -616,19 +611,43 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
           this.dispatchEvent(new shaka.util.FakeEvent(
               'error', (new Map()).set('detail', error)));
         }
+      } else {
+        const video = /** @type {HTMLVideoElement} */(this.localVideo_);
+        if (video.webkitSupportsFullscreen) {
+          video.webkitEnterFullscreen();
+        }
+      }
+    } catch (error) {
```

Comment:
There are three nested try/catch blocks.  I think that might be unnecessary and confusing.

With your current code, we catch these errors and respond in these ways:

1. exit PiP fails: dispatch error event (no explanation)
2. standard FS API fails: dispatch error event (no explanation)
3. orientation lock fails: ignore (may not be supported on device)
4. webkit FS API fails: ignore (may not work without user interaction)

I think this error handling is inconsistent and deserves to be rethought.

Should we suppress all errors here?  These are the reasons the standard FS API might fail: https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullscreen#exceptions

If we're not suppressing all errors, perhaps we should dispatch an error event for any case where the FS operation fails, including using the webkit API.  It might be easier for apps to understand things like "why did my video not go fullscreen on iOS" if there's an error event with a hint.